### ISSUE TITLE
validateColorSpaceArray: undefined color space: DeviceRGB

### ISSUE THREAD

**Issue by aidansteele** [pdfcpu/pdfcpu#549]
pdfcpu version: 0.3.13 (latest as of writing)
OS: macOS Ventura 13.1

<details>
  <summary>Crash log (click to expand)</summary>

```
validating(mode=relaxed) input.pdf ...
 INFO: 2023/01/21 10:41:54 PDF Version 1.5 conforming reader
 INFO: 2023/01/21 10:41:54 validating
STATS: 2023/01/21 10:41:54 XRefTable:
*************************************************************************************************
HeaderVersion: 1.5
has 2 pages
using object streams
using xref streams
is linearized file
XRefTable:
                     Size: 52
              Root object: (30 0 R)
              Info object: (28 0 R)
                ID object: [<7124AC8D0CC29F12C31DDEF78DA13C0E> <ECFE4ED5C692C64D885A5C0B52314B9D>]
XRefTable with 52 entries:
    0: f   next=       0 generation=65535
    1:   offset=  282333 generation=0 pdfcpu.Dict type=Page 
<<
	<Contents, (2 0 R)>
	<CropBox, [0 0 595.24 841.84]>
	<MediaBox, [0 0 595.24 841.84]>
	<Parent, (27 0 R)>
	<Resources, <<
		<ExtGState, <<
			<GS1, (45 0 R)>
		>>>
		<Font, <<
			<Font-0, (48 0 R)>
			<Font-1, (24 0 R)>
			<HeBo, (49 0 R)>
		>>>
		<ProcSet, [PDF Text ImageC ImageB ImageI]>
		<XObject, <<
			<im5289, (5 0 R)>
			<im5290, (6 0 R)>
			<im5291, (7 0 R)>
			<im5292, (8 0 R)>
			<im5293, (9 0 R)>
			<im5294, (10 0 R)>
			<im5295, (11 0 R)>
			<im5296, (12 0 R)>
			<im5297, (13 0 R)>
			<im5298, (14 0 R)>
		>>>
	>>>
	<Rotate, 0>
	<Type, Page>
>>
    2:   offset=  282735 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 3582>
>>
    3:   offset=  286387 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 2218>
	<Length1, 2904>
>>
    4:   offset=  288688 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 347>
>>
    5:   offset=  289104 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 44>
	<Interpolate, true>
	<Length, 5371>
	<Subtype, Image>
	<Type, XObject>
	<Width, 294>
>>
    6:   offset=  294647 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 2400>
	<Interpolate, true>
	<Length, 871364>
	<Subtype, Image>
	<Type, XObject>
	<Width, 3600>
>>
    7:   offset= 1166188 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, [DeviceRGB]>
	<Filter, FlateDecode>
	<Height, 55>
	<Interpolate, true>
	<Length, 371>
	<Subtype, Image>
	<Type, XObject>
	<Width, 70>
>>
    8:   offset= 1166733 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, [DeviceRGB]>
	<Filter, FlateDecode>
	<Height, 55>
	<Interpolate, true>
	<Length, 392>
	<Subtype, Image>
	<Type, XObject>
	<Width, 70>
>>
    9:   offset= 1167299 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, [DeviceRGB]>
	<Filter, FlateDecode>
	<Height, 55>
	<Interpolate, true>
	<Length, 799>
	<Subtype, Image>
	<Type, XObject>
	<Width, 70>
>>
   10:   offset= 1168272 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 120>
	<Interpolate, true>
	<Length, 7743>
	<Subtype, Image>
	<Type, XObject>
	<Width, 160>
>>
   11:   offset= 1176189 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 120>
	<Interpolate, true>
	<Length, 8596>
	<Subtype, Image>
	<Type, XObject>
	<Width, 160>
>>
   12:   offset= 1184959 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 107>
	<Interpolate, true>
	<Length, 7062>
	<Subtype, Image>
	<Type, XObject>
	<Width, 160>
>>
   13:   offset= 1192195 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 265>
	<Interpolate, true>
	<Length, 20344>
	<Subtype, Image>
	<Type, XObject>
	<Width, 493>
>>
   14:   offset= 1212714 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 50>
	<Interpolate, true>
	<Length, 3142>
	<Subtype, Image>
	<Type, XObject>
	<Width, 180>
>>
   15:   offset= 1216029 generation=0 pdfcpu.ObjectStreamDict 
<<
	<Filter, FlateDecode>
	<First, 12>
	<Length, 347>
	<N, 2>
	<Type, ObjStm>
>>
object stream count:2 size of objectarray:2
   16:   offset= 1216471 generation=0 pdfcpu.ObjectStreamDict 
<<
	<Filter, FlateDecode>
	<First, 5>
	<Length, 128>
	<N, 1>
	<Type, ObjStm>
>>
object stream count:1 size of objectarray:1
   17:   offset= 1216693 generation=0 pdfcpu.ObjectStreamDict 
<<
	<Filter, FlateDecode>
	<First, 5>
	<Length, 19>
	<N, 1>
	<Type, ObjStm>
>>
object stream count:1 size of objectarray:1
   18:   offset= 1216805 generation=0 pdfcpu.StreamDict 
<<
	<Length, 3655>
	<Subtype, XML>
	<Type, Metadata>
>>
   19:   offset= 1220538 generation=0 pdfcpu.ObjectStreamDict 
<<
	<Filter, FlateDecode>
	<First, 11>
	<Length, 91>
	<N, 2>
	<Type, ObjStm>
>>
object stream count:2 size of objectarray:2
   20:   offset= 1220723 generation=0 pdfcpu.ObjectStreamDict 
<<
	<Filter, FlateDecode>
	<First, 5>
	<Length, 197>
	<N, 1>
	<Type, ObjStm>
>>
object stream count:1 size of objectarray:1
   21:   offset= 1221014 generation=0 pdfcpu.XRefStreamDict 
<<
	<DecodeParms, <<
		<Columns, 5>
		<Predictor, 12>
	>>>
	<Filter, FlateDecode>
	<ID, [<7124AC8D0CC29F12C31DDEF78DA13C0E> <ECFE4ED5C692C64D885A5C0B52314B9D>]>
	<Info, (28 0 R)>
	<Length, 107>
	<Root, (30 0 R)>
	<Size, 29>
	<Type, XRef>
	<W, [1 3 1]>
>>
   22: was compressed 15[0] generation=0 pdfcpu.Dict type=FontDescriptor 
<<
	<Ascent, 940>
	<CapHeight, 715>
	<Descent, -260>
	<Flags, 32>
	<FontBBox, [-118 -210 1088 883]>
	<FontFile2, (3 0 R)>
	<FontName, SUBSET+MuseoSans300>
	<ItalicAngle, 0>
	<MissingWidth, 1000>
	<StemV, 0>
	<Type, FontDescriptor>
>>
   23: was compressed 15[1] generation=0 pdfcpu.Dict type=Font subType=CIDFontType2 
<<
	<BaseFont, SUBSET+MuseoSans300>
	<CIDSystemInfo, <<
		<Ordering, (UCS)>
		<Registry, (Adobe)>
		<Supplement, 0>
	>>>
	<CIDToGIDMap, Identity>
	<DW, 500>
	<FontDescriptor, (22 0 R)>
	<Name, Font-1>
	<Subtype, CIDFontType2>
	<Type, Font>
	<W, [0 [0 595 625 565 392 597 558 618 232 549 335 622 356 520 598 589 553 244 728 247 593 582 801 562 916 589 589 443]]>
>>
   24: was compressed 16[0] generation=0 pdfcpu.Dict type=Font subType=Type0 
<<
	<BaseFont, SUBSET+MuseoSans300>
	<DescendantFonts, [(23 0 R)]>
	<Encoding, Identity-H>
	<Name, Font-1>
	<Subtype, Type0>
	<ToUnicode, (4 0 R)>
	<Type, Font>
>>
   25: was compressed 17[0] generation=0 pdfcpu.Dict 
<<
>>
   26: was compressed 19[0] generation=0 pdfcpu.Dict type=Pages 
<<
	<Count, 2>
	<ITXT, (5.1.3)>
	<Kids, [(27 0 R)]>
	<Type, Pages>
>>
   27: was compressed 19[1] generation=0 pdfcpu.Dict type=Pages 
<<
	<Count, 2>
	<Kids, [(31 0 R) (1 0 R)]>
	<Parent, (26 0 R)>
	<Type, Pages>
>>
   28: was compressed 20[0] generation=0 pdfcpu.Dict 
<<
	<Author, (PropertyData Online)>
	<CreationDate, (D:20220511145541Z)>
	<ModDate, (D:20220511145719+10'00')>
	<Producer, (xmlpdf 5.5.4M [.NET 1.0]; modified using iTextSharp 5.1.3 \(c\) 1T3XT BVBA)>
	<Subject, (PropertyData Online - Property Report)>
	<Title, (PropertyData Online - Property Report)>
>>
   29:   offset=      16 generation=0 pdfcpu.Dict 
<<
	<E, 282333>
	<H, [483 209]>
	<L, 1221377>
	<Linearized, 1>
	<N, 2>
	<O, 31>
	<T, 1221013>
>>
   30:   offset=     692 generation=0 pdfcpu.Dict type=Catalog 
<<
	<Metadata, (18 0 R)>
	<Outlines, (25 0 R)>
	<PageMode, UseNone>
	<Pages, (26 0 R)>
	<Type, Catalog>
>>
   31:   offset=     788 generation=0 pdfcpu.Dict type=Page 
<<
	<Contents, [(32 0 R) (33 0 R) (36 0 R) (37 0 R) (38 0 R) (39 0 R) (40 0 R) (41 0 R)]>
	<CropBox, [0 0 595.24 841.84]>
	<MediaBox, [0 0 595.24 841.84]>
	<Parent, (27 0 R)>
	<Resources, <<
		<ExtGState, <<
			<GS1, (45 0 R)>
		>>>
		<Font, <<
			<Font-0, (48 0 R)>
			<HeBo, (49 0 R)>
			<TiRo, (50 0 R)>
		>>>
		<ProcSet, [PDF Text ImageC ImageB ImageI]>
		<XObject, <<
			<im5288, (43 0 R)>
		>>>
	>>>
	<Rotate, 0>
	<Type, Page>
>>
   32:   offset=    1119 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 570>
>>
   33:   offset=    1759 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 636>
>>
   34:   offset=    2465 generation=0 pdfcpu.ObjectStreamDict 
<<
	<Filter, FlateDecode>
	<First, 39>
	<Length, 525>
	<N, 6>
	<Type, ObjStm>
>>
object stream count:6 size of objectarray:6
   35:   offset=    3085 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 502>
>>
   36:   offset=    3657 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 721>
>>
   37:   offset=    4448 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 933>
>>
   38:   offset=    5451 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 852>
>>
   39:   offset=    6373 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 754>
>>
   40:   offset=    7197 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 578>
>>
   41:   offset=    7845 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 693>
>>
   42:   offset=    8608 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<Length, 8006>
	<Length1, 13476>
>>
   43:   offset=   16699 generation=0 pdfcpu.StreamDict 
<<
	<BitsPerComponent, 8>
	<ColorSpace, DeviceRGB>
	<Filter, DCTDecode>
	<Height, 1181>
	<Interpolate, true>
	<Length, 265456>
	<Subtype, Image>
	<Type, XObject>
	<Width, 4257>
>>
   44:   offset=     116 generation=0 pdfcpu.XRefStreamDict 
<<
	<DecodeParms, <<
		<Columns, 4>
		<Predictor, 12>
	>>>
	<Filter, FlateDecode>
	<ID, [<7124AC8D0CC29F12C31DDEF78DA13C0E> <ECFE4ED5C692C64D885A5C0B52314B9D>]>
	<Index, [29 23]>
	<Info, (28 0 R)>
	<Length, 79>
	<Prev, 1221014>
	<Root, (30 0 R)>
	<Size, 52>
	<Type, XRef>
	<W, [1 2 1]>
>>
   45: was compressed 34[0] generation=0 pdfcpu.Dict type=ExtGState 
<<
	<LC, 0>
	<LJ, 0>
	<SA, true>
	<Type, ExtGState>
>>
   46: was compressed 34[1] generation=0 pdfcpu.Dict type=FontDescriptor 
<<
	<Ascent, 952>
	<CapHeight, 715>
	<Descent, -213>
	<Flags, 32>
	<FontBBox, [-348 -214 1086 952]>
	<FontFile2, (42 0 R)>
	<FontName, SUBSET+HelveticaNeue>
	<ItalicAngle, 0>
	<MissingWidth, 1000>
	<StemV, 0>
	<Type, FontDescriptor>
>>
   47: was compressed 34[2] generation=0 pdfcpu.Dict type=Font subType=CIDFontType2 
<<
	<BaseFont, SUBSET+HelveticaNeue>
	<CIDSystemInfo, <<
		<Ordering, (UCS)>
		<Registry, (Adobe)>
		<Supplement, 0>
	>>>
	<CIDToGIDMap, Identity>
	<DW, 500>
	<FontDescriptor, (46 0 R)>
	<Name, Font-0>
	<Subtype, CIDFontType2>
	<Type, Font>
	<W, [0 [0 648 315 537 537 853 556 574 296 259 333 222 648 593 500 537 222 556 574 593 593 556 556 556 333 556 556 611 648 278 722 758 556 574 556 278 500 593 556 556 871 556 500 574 722 389 556 685 611 519 518 278 760 556 722 759 926 704 556 278 278 222 259 259]]>
>>
   48: was compressed 34[3] generation=0 pdfcpu.Dict type=Font subType=Type0 
<<
	<BaseFont, SUBSET+HelveticaNeue>
	<DescendantFonts, [(47 0 R)]>
	<Encoding, Identity-H>
	<Name, Font-0>
	<Subtype, Type0>
	<ToUnicode, (35 0 R)>
	<Type, Font>
>>
   49: was compressed 34[4] generation=0 pdfcpu.Dict type=Font subType=Type1 
<<
	<BaseFont, Helvetica-Bold>
	<Encoding, WinAnsiEncoding>
	<Name, HeBo>
	<Subtype, Type1>
	<Type, Font>
>>
   50: was compressed 34[5] generation=0 pdfcpu.Dict type=Font subType=Type1 
<<
	<BaseFont, Times-Roman>
	<Encoding, WinAnsiEncoding>
	<Name, TiRo>
	<Subtype, Type1>
	<Type, Font>
>>
   51:   offset=     483 generation=0 pdfcpu.StreamDict 
<<
	<Filter, FlateDecode>
	<I, 132>
	<Length, 122>
	<O, 116>
	<S, 56>
>>

Empty free list.

Total pages: 2
No font info available.

No image info available.


STATS: 2023/01/21 10:41:54 Timing:
STATS: 2023/01/21 10:41:54 read                 :  0.002s  92.1%
STATS: 2023/01/21 10:41:54 validate             :  0.000s   7.9%
STATS: 2023/01/21 10:41:54 total processing time:  0.002s

STATS: 2023/01/21 10:41:54 Original:
STATS: 2023/01/21 10:41:54 File size            : 1.2 MB (1221377 bytes)
STATS: 2023/01/21 10:41:54 Total binary data    : 1.2 MB (1216302 bytes) 99.6%
STATS: 2023/01/21 10:41:54 Total other data     : 5 KB (5075 bytes)  0.4%

Fatal: pdfcpu: validateColorSpaceArray: undefined color space: DeviceRGB

github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateColorSpaceArray
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/colorspace.go:560
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateColorSpaceEntry
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/colorspace.go:607
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateImageStreamDictPart1
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/xObject.go:447
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateImageStreamDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/xObject.go:517
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateXObjectStreamDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/xObject.go:801
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateXObjectResourceDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/xObject.go:868
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateResourceDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/pages.go:45
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validatePageResources
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/pages.go:114
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validatePageDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/pages.go:773
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validatePagesDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/pages.go:993
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validatePagesDict
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/pages.go:988
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validatePages
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/pages.go:1032
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateRootObject
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/xReftable.go:943
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.XRefTable
	github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate/xReftable.go:40
github.com/pdfcpu/pdfcpu/pkg/api.ValidateContext
	github.com/pdfcpu/pdfcpu/pkg/api/api.go:70
github.com/pdfcpu/pdfcpu/pkg/api.Validate
	github.com/pdfcpu/pdfcpu/pkg/api/validate.go:51
github.com/pdfcpu/pdfcpu/pkg/api.ValidateFile
	github.com/pdfcpu/pdfcpu/pkg/api/validate.go:92
github.com/pdfcpu/pdfcpu/pkg/api.ValidateFiles
	github.com/pdfcpu/pdfcpu/pkg/api/validate.go:115
github.com/pdfcpu/pdfcpu/pkg/cli.Validate
	github.com/pdfcpu/pdfcpu/pkg/cli/cli.go:32
github.com/pdfcpu/pdfcpu/pkg/cli.Process
	github.com/pdfcpu/pdfcpu/pkg/cli/process.go:35
main.process
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/process.go:102
main.processValidateCommand
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/process.go:157
main.commandMap.process
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/cmd.go:143
main.main
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/main.go:55
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_arm64.s:1172
validation error (obj#:22)
github.com/pdfcpu/pdfcpu/pkg/api.Validate
	github.com/pdfcpu/pdfcpu/pkg/api/validate.go:56
github.com/pdfcpu/pdfcpu/pkg/api.ValidateFile
	github.com/pdfcpu/pdfcpu/pkg/api/validate.go:92
github.com/pdfcpu/pdfcpu/pkg/api.ValidateFiles
	github.com/pdfcpu/pdfcpu/pkg/api/validate.go:115
github.com/pdfcpu/pdfcpu/pkg/cli.Validate
	github.com/pdfcpu/pdfcpu/pkg/cli/cli.go:32
github.com/pdfcpu/pdfcpu/pkg/cli.Process
	github.com/pdfcpu/pdfcpu/pkg/cli/process.go:35
main.process
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/process.go:102
main.processValidateCommand
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/process.go:157
main.commandMap.process
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/cmd.go:143
main.main
	github.com/pdfcpu/pdfcpu/cmd/pdfcpu/main.go:55
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_arm64.s:1172
```

</details>

I've attached the PDF that causes this error: [input.pdf](https://github.com/pdfcpu/pdfcpu/files/10471001/input.pdf)

Thanks for a wonderful library!

**Comment by petervwyatt** [comment]
Your PDF is technically malformed according to the PDF specifications and standards: it is the difference between `/ColorSpace /DeviceRGB` (correct) and `/ColorSpace [ /DeviceRGB ]` (incorrect) for Image XObjects. Your file uses both the correct and incorrect syntax so something is wrong with either the original PDF or the tool used to add the incremental update (Prodcuer metadata is `xmlpdf 5.5.4M [.NET 1.0]; modified using iTextSharp 5.1.3 (c) 1T3XT BVBA`). 

Rewriting the entire PDF rather than using incremental updates will likely correct the malformations no matter who caused it.

**Comment by hhrutter** [comment]
There is at least one invalid ColorSpace Array in your file since `DeviceRGB` is supposed to be declared as a Name Object and therefore invalid if it appears withing a Colorspace Array.

I'll see if extending relaxed validation makes sense and gets you any further processing this.
Stay tuned for the upcoming release.

**Comment by hhrutter** [comment]
This should be fixed with v0.4.0
